tests/js/server/aql/aql-arangosearch-count-optimization.js

```diff
@@ -0,0 +1,171 @@
+/*jshint globalstrict:false, strict:false, maxlen: 500 */
+/*global assertTrue, assertFalse, assertEqual, assertNotEqual, AQL_EXECUTE, AQL_EXPLAIN */
+
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2020 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Andrei Lobov
+/// @author Copyright 2020, ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+// Some tests skips cluster until ticket TG-1110 (Some fullCount seems to not be propagated well enough)
+// is fixed. Or all cluster tests with fullCount will hang
+
+var jsunity = require("jsunity");
+var internal = require("internal");
+var errors = internal.errors;
+var db = require("@arangodb").db;
+var helper = require("@arangodb/aql-helper");
+var assertQueryError = helper.assertQueryError;
+const isCluster = require("@arangodb/cluster").isCluster();
+
+function arangoSearchCountOptimization () {
+  return {
+    setUpAll : function () {
+      db._dropView("UnitTestView");
+      db._dropView("UnitTestViewSorted");
+      db._drop("UnitTestsCollection");
+      let c = db._create("UnitTestsCollection", { numberOfShards: 3 });
+
+      for (let i = 0; i < 10010; ++i) {
+        c.save({ value: "test" + (i % 10), count: i });
+      }
```

Comment:
Turn 10k HTTP requests in (a big) one
```suggestion
      let docs = [];
      for (let i = 0; i < 10010; ++i) {
        docs.push({ value: "test" + (i % 10), count: i });
      }
      c.insert(docs);
```